Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import MapLibre
+import MapLibreSwiftDSL
+
+public class MapViewCoordinator: NSObject {
+    
+    // This must be weak, the UIViewRepresentable owns the MLNMapView.
+    weak var mapView: MLNMapView?
+    var parent:  MapView
+
+    // Storage of variables as they were previously; these are snapshot
+    // every update cycle so we can avoid unnecessary updates
+    private var snapshotUserLayers: [StyleLayerDefinition] = []
+    private var snapshotCamera: MapViewCamera?
+    private var onGestureEnd: (MLNMapView, UIGestureRecognizer) -> Void
+    
+    init(parent: MapView,
+         onGestureEnd: @escaping (MLNMapView, UIGestureRecognizer) -> Void) {
+        self.parent = parent
+        self.onGestureEnd = onGestureEnd
+    }
+    
+    // MARK: Core UIView Functionality
+    
+    @objc func captureGesture(_ sender: UIGestureRecognizer) {
+        guard let mapView, sender.state == .ended else {
```

Comment:
Hinted at elsewhere, but do we also want to add support for started gestures while we're in here just so we've got the scaffolding in place for future extension (don't need to add all cases now, but maybe two is helpful to set the patterns)?

Reply:
I revamped the gesture system entirely to allow any number of custom configurations. Our discussion was an excellent kick off to think through a better way. Mainly, the chat about rust's `Arc` -> duh, let's just capture the gesture recognizer class and check if the instance is the right one.